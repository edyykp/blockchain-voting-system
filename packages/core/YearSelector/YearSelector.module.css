.container {
  display: flex;
  align-items: center;
  justify-content: center;
}

.leftLine {
  width: 19%;
  background-color: black;
  height: 1px;
  border-width: 0;
  -webkit-box-shadow: 2px 2px 10px 0px #666666;
  box-shadow: 2px 2px 10px 0px #666666;
}

.rightLine {
  width: 70%;
  height: 1px;
  border-width: 0;
  background-color: black;
  -webkit-box-shadow: 2px 2px 10px 0px #666666;
  box-shadow: 2px 2px 10px 0px #666666;
}

.selector {
  height: 30px;
  width: 10%;
  font-size: 1rem;
  border-bottom: 1px solid black;
  border: 0;
  background-color: var(--gray-800);
  color: var(--gray-200);
  opacity: 0.7;
  -webkit-box-shadow: 2px 2px 10px 0px #666666;
  box-shadow: 2px 2px 10px 0px #666666;
  border-radius: 5px;
}

@media screen and (max-width: 1024px) {
  .selector {
    width: 15%;
  }

  .leftLine {
    width: 17%;
  }

  .rightLine {
    width: 67%;
  }
}

@media screen and (max-width: 768px) {
  .selector {
    width: 18%;
  }

  .leftLine {
    width: 15%;
  }

  .rightLine {
    width: 66%;
  }
}

@media screen and (max-width: 520px) {
  .selector {
    width: 30%;
  }

  .leftLine {
    width: 11%;
  }

  .rightLine {
    width: 59%;
  }
}
