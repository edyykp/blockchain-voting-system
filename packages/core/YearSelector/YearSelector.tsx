import { useRouter } from 'next/router';

const currentYear = new Date().getFullYear();
const listOfYears: number[] = [];
for (let year = 2015; year <= currentYear; year++) {
  listOfYears.push(year);
}

export const YearSelector = () => {
  const router = useRouter();

  return (
    <select
      name="years"
      id="years"
      onChange={(event) =>
        router.replace(`/dashboard?year=${event.target.value}`)
      }
      defaultValue={router.query.year ?? currentYear}
      data-testid="year-selector"
    >
      {listOfYears.map((year) => (
        <option value={year} key={year}>
          {year}
        </option>
      ))}
    </select>
  );
};
