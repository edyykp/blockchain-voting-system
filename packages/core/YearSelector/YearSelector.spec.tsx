import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { YearSelector } from './YearSelector';

const setup = (year: string) => {
  return {
    ...render(<YearSelector year={year} />),
  };
};

describe('YearSelectorâ†’', () => {
  it('renders', () => {
    const { container, getByTestId } = setup('2017');

    expect(getByTestId('year-selector')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('renders correct year as selected', () => {
    const { getByRole } = setup('2019');

    expect(
      getByRole('option', { name: 'Year 2019' }).getAttribute('selected'),
    ).toBe('');
    expect(
      getByRole('option', { name: 'Year 2022' }).getAttribute('selected'),
    ).toBeNull();
  });
});
