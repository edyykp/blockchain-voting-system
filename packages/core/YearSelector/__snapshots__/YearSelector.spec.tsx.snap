// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YearSelectorâ†’ renders 1`] = `
<div>
  <span
    class="container"
  >
    <hr
      class="leftLine"
    />
    <select
      class="selector"
      data-testid="year-selector"
      id="years"
      name="years"
    >
      <option
        class="option"
        value="2015"
      >
        Year 2015
      </option>
      <option
        class="option"
        value="2016"
      >
        Year 2016
      </option>
      <option
        class="option"
        selected=""
        value="2017"
      >
        Year 2017
      </option>
      <option
        class="option"
        value="2018"
      >
        Year 2018
      </option>
      <option
        class="option"
        value="2019"
      >
        Year 2019
      </option>
      <option
        class="option"
        value="2020"
      >
        Year 2020
      </option>
      <option
        class="option"
        value="2021"
      >
        Year 2021
      </option>
      <option
        class="option"
        value="2022"
      >
        Year 2022
      </option>
      <option
        class="option"
        value="2023"
      >
        Year 2023
      </option>
    </select>
    <hr
      class="rightLine"
    />
  </span>
</div>
`;
